### PR TITLE
Replaces Commons Logging usage with Slf4j - Fixes #613

### LINKED ISSUE #613: Replace Commons Logging usage with Slf4j and review logging output
As of #612, we are officially using logback logging rather than legacy log4j 1.  Now we need to address switching over our own internal logging to use slf4j so that we are not wrapped through jcl-to-slf4j needlessly in order to improve performance.  Additionally, this should be an opportunately to review logging configuration in case any adjustments are necessary.  The basic configuration as it stands is similar to that of tomcat logging at this time.

### PR DESCRIPTION
**Pull request by diogosantana**
Some loggers atributes were added in classes that don't use it. The
reason is to prevent usage of Spring super classes protected logger
atribute.

@hazendaz can you check if the line endings problem is occurring?

### CODE REVIEW

**Comment by hazendaz** [comment]
Should be fine.  So far I've only seen line ending issues on new files.


Sent from Outlook Mobile






On Sun, Feb 7, 2016 at 1:41 AM -0800, "diogosantana" <notifications@github.com> wrote:





Some loggers atributes were added in classes that don't use it. The
reason is to prevent usage of Spring super classes protected logger
atribute.

@hazendaz can you check if the line endings problem is occurring?
You can view, comment on, or merge this pull request online at:

  https://github.com/psi-probe/psi-probe/pull/656

-- Commit Summary --

  * Replaces Commons Logging usage with Slf4j - Fixes #613

-- File Changes --

    M core/src/main/java/psiprobe/AbstractTomcatContainer.java (2)
    M core/src/main/java/psiprobe/ProbeServlet.java (5)
    M core/src/main/java/psiprobe/controllers/ContextHandlerController.java (5)
    M core/src/main/java/psiprobe/controllers/RenderChartController.java (5)
    M core/src/main/java/psiprobe/controllers/TomcatContainerController.java (5)
    M core/src/main/java/psiprobe/controllers/WhoisController.java (5)
    M core/src/main/java/psiprobe/controllers/apps/AjaxReloadContextController.java (2)
    M core/src/main/java/psiprobe/controllers/apps/AjaxToggleContextController.java (2)
    M core/src/main/java/psiprobe/controllers/apps/NoSelfContextHandlerController.java (2)
    M core/src/main/java/psiprobe/controllers/deploy/UndeployContextController.java (2)
    M core/src/main/java/psiprobe/controllers/deploy/UploadWarController.java (2)
    M core/src/main/java/psiprobe/controllers/sql/CachedRecordSetController.java (5)
    M core/src/main/java/psiprobe/controllers/sql/QueryHistoryItemController.java (5)
    M core/src/main/java/psiprobe/controllers/system/AdviseGarbageCollectionController.java (5)
    M core/src/main/java/psiprobe/controllers/threads/GetClassLoaderUrlsController.java (5)
    M core/src/main/java/psiprobe/controllers/wrapper/RestartJvmController.java (5)
    M core/src/main/java/psiprobe/controllers/wrapper/StopJvmController.java (5)
    M core/src/main/java/psiprobe/controllers/wrapper/ThreadDumpController.java (5)
    M core/src/main/java/psiprobe/controllers/wrapper/WrapperInfoController.java (5)

-- Patch Links --

https://github.com/psi-probe/psi-probe/pull/656.patch
https://github.com/psi-probe/psi-probe/pull/656.diff

---
Reply to this email directly or view it on GitHub:
https://github.com/psi-probe/psi-probe/pull/656

File: core/src/main/java/psiprobe/AbstractTomcatContainer.java [hunk 1]
```diff
@@ -406,7 +406,7 @@ public void listContextJsps(Context context, Summary summary, boolean compile) {
   }
 
   @Override
-  public org.apache.juli.logging.Log getLogger(Context context) {
+  public Object getLogger(Context context) {
```
Comment (MALfunction84) [inline core/src/main/java/psiprobe/AbstractTomcatContainer.java]: What motivated this change? All supported Tomcat versions return `org.apache.juli.logging.Log`.

Comment (hazendaz) [inline core/src/main/java/psiprobe/AbstractTomcatContainer.java]: Main objective is to make sure we use our own loggers and don't mistakening use tomcats.  In this case, not once is this method ever used other than as Object.  However, I see zero point for this in the interface or the concrete class.  It's 3 use cases simply call through the wrapper very pointlessly passing 'context' which then just reads context.getLogger().  The wrapper calls provide no additional processing and in my opinion are not needed.  Instead we should delete the interface method, concrete method in abstract class, and then just call it directly thus not wasting time calling through objects.

After deleting those, in LogResolverBean, the 3 calls all will be the same.

Previous
```
Object contextLogger = getContainerWrapper().getTomcatContainer().getLogger(ctx);
```
New would still just receive object back because it can be considered different in a couple of the calls plus setting wants Object not actual log type.
```
Object contextLogger = ctx.getLogger();
```
Please take a quick look over this to see if I'm missing anything in this actual use case.  I've got this switched locally as my thinking suggests above.  I'll be running some tests shortly.

Comment (MALfunction84) [inline core/src/main/java/psiprobe/AbstractTomcatContainer.java]: Most likely a holdover from an older version of Tomcat. What you're saying makes sense. Let's remove it.

Comment (hazendaz) [inline core/src/main/java/psiprobe/AbstractTomcatContainer.java]: Is it ok to do that on a separate Pull Request?  I don't see much harm letthing this one through then following up with that fix.

**Comment by hazendaz** [comment]
OK - no further comment.  I'm merging as fix will come later to remove the unnecessary method and I'm trying to apply proper usage of slf4j without string concat.  There are a few incorrect here so I'll pick those up on my branch.  Loggers should be static final but that can be addressed later.  The main objective here is completed.  Sonar will notify us on the rest :)
